### ISSUE TITLE
fig under jenkins problems

### ISSUE THREAD

**Issue by ccverak** [docker/compose#199]
It does not flushes logs and has encoding issues when running on jenkins

**Comment by tobowers** [comment]
I'm still seeing this with docker-compose 1.1.0 (well something similar). I see the first line of stdout, but not anything after the first newline:

```
sudo docker-compose run web /bin/bash -c 'export PYTHONUNBUFFERED=1 && echo "hi" && echo "hi" && echo "hi"'
Creating rspec19_db_1...
hi
# nothing after here
```

**Comment by mattupstate** [comment]
@tobowers as am I

**Comment by tobowers** [comment]
I just used -T and it *seemed* like it helped, but didn't fix it. Seems like there are a couple of outstanding issues from january: https://github.com/docker/compose/issues/658 kind of sums it up @mattupstate

**Comment by mattupstate** [comment]
@tobowers Yeah, I've been aware of this for a little while now. `-T` never worked perfectly for me under Jenkins even after it was thought to be fixed.

**Comment by zwily** [comment]
This is also still not working for me. Based on what everyone's saying, it seems like the issue should still be open.

**Comment by dnephin** [comment]
It does appear like our earlier attempt did not completely fix this issue.

**Comment by suryagaddipati** [comment]
I am seeing this as well. It seems to dropping the output when there is a pause in the output.

**Comment by arthurk** [comment]
same here, the issue is still there. I'm running docker-compose 1.2.0 and Jenkins 1.6.12

**Comment by aanand** [comment]
If anyone's having truncation issues, you could try the binaries built by @zwily which contain the fix from https://github.com/d11wtq/dockerpty/pull/36:

https://github.com/docker/compose/issues/658#issuecomment-96443431

**Comment by dnephin** [comment]
This fix has been merged, and we've bumped the version of `dockerpty` used by compose. This fix is available in master, and will be inlcuded in the 1.3.0 release.
